### PR TITLE
Prototype of Embedded etcd

### PR DESCRIPTION
**Pull request by JackyChiu**
# What
We've had the idea of using an etcd cluster to manage our service discovery and kv store. The issue was that for our platform we didn't want to rely on an external cluster, the developers should simply be able to use their own nodes for this. The embedded etcd package allows us to do just that. 

# Bootstrapping
More details on the configuration can be seen here: https://github.com/etcd-io/etcd/blob/master/Documentation/op-guide/configuration.md

The TLDR version is that `peer_urls`s are other etcd nodes, and `client_urls` are etcd clients who want to interact with the kv store. The bootstrap config is anything that starts with `intial`. 

# Run
```
ETCD_CONFIG=config/node1.yml go run cmd/ptype/main.go
ETCD_CONFIG=config/node2.yml go run cmd/ptype/main.go
ETCD_CONFIG=config/node3.yml go run cmd/ptype/main.go
```

### CODE REVIEW

**Comment by vikrambombhi** [comment]
We should try and figure out the minimal set of configuration options and make things dynamic during runtime

**Comment by JackyChiu** [comment]
> We should try and figure out the minimal set of configuration options and make things dynamic during runtime

I have an idea of the minimal set, I would need to try it out to verify tho. Do we want to make this a priority for this week? We currently don't have any others so I'll add this to the board.
